pkg/types/symboltable/symboltable.go

```diff
@@ -0,0 +1,419 @@
+
+
+func (sym *SymbolDef) GetPosition() DeclPos { return sym.Pos }
+func (sym *SymbolDef) GetName() string      { return sym.Name }
+func (sym *SymbolDef) GetType() string      { return sym.Def.GetType() }
+
+//TODO: maybe something like this would be part of types.go? Used just
+//      to find data type of processed symbol
+//TODO: signature... would be fine use same as json.Unmarshal
+func unmarshalDataType(b []byte) (gotypes.DataType, error) {
+	var objMap map[string]*json.RawMessage
+
+	if err := json.Unmarshal(b, &objMap); err != nil {
+		return nil, err
+	}
+
+	var symbolType string
+	if err := json.Unmarshal(*objMap["type"], &symbolType); err != nil {
+		return nil, err
+	}
+
+	var symbol gotypes.DataType
+	switch symbolType {
+	case gotypes.FunctionType:
+		var t gotypes.Function
+		if err := json.Unmarshal(b, &t); err != nil {
+			return nil, err
+		}
+		symbol = &t
+	case gotypes.MapType:
+		var t gotypes.Map
+		if err := json.Unmarshal(b, &t); err != nil {
+			return nil, err
+		}
+		symbol = &t
+	case gotypes.SliceType:
+		var t gotypes.Slice
+		if err := json.Unmarshal(b, &t); err != nil {
+			return nil, err
+		}
+		symbol = &t
+	case gotypes.StructFieldsItemType:
+		var t gotypes.StructFieldsItem
+		if err := json.Unmarshal(b, t); err != nil {
+			return nil, err
+		}
+		symbol = &t
+	case gotypes.StructType:
+		var t gotypes.Struct
+		if err := json.Unmarshal(b, &t); err != nil {
+			return nil, err
+		}
+		symbol = &t
+	case gotypes.SelectorType:
+		var t gotypes.Selector
+		if err := json.Unmarshal(b, &t); err != nil {
+			return nil, err
+		}
+		symbol = &t
+	case gotypes.InterfaceMethodsItemType:
+		var t gotypes.InterfaceMethodsItem
+		if err := json.Unmarshal(b, &t); err != nil {
+			return nil, err
+		}
+		symbol = &t
+	case gotypes.InterfaceType:
+		var t gotypes.Interface
+		if err := json.Unmarshal(b, &t); err != nil {
+			return nil, err
+		}
+		symbol = &t
+	case gotypes.EllipsisType:
+		var t gotypes.Ellipsis
+		if err := json.Unmarshal(b, &t); err != nil {
+			return nil, err
+		}
+		symbol = &t
+	case gotypes.ArrayType:
+		var t gotypes.Array
+		if err := json.Unmarshal(b, &t); err != nil {
+			return nil, err
+		}
+		symbol = &t
+	case gotypes.IdentifierType:
+		var t gotypes.Identifier
+		if err := json.Unmarshal(b, &t); err != nil {
+			return nil, err
+		}
+		symbol = &t
+	case gotypes.PointerType:
+		var t gotypes.Pointer
+		if err := json.Unmarshal(b, &t); err != nil {
+			return nil, err
+		}
+		symbol = &t
+	case gotypes.MethodType:
+		var t gotypes.Method
+		if err := json.Unmarshal(b, &t); err != nil {
+			return nil, err
+		}
+		//FIXME: methods and hash-names - see prefix
+		symbol = &t
+	case gotypes.ChannelType:
+		var t gotypes.Channel
+		if err := json.Unmarshal(b, &t); err != nil {
+			return nil, err
+		}
+		symbol = &t
+	default:
+		return nil, fmt.Errorf("JSON unmarshal: Unknown symbol type '%v'", symbolType)
+	}
+
+	return symbol, nil
+}
+
+func (o *SymbolDef) UnmarshalJSON(b []byte) error {
```

Comment:
this is already generated https://github.com/gofed/symbols-extractor/commit/6223b32617476e16b5dcb402af5513236cee0fc5